### PR TITLE
Add simple runscript sanity test

### PR DESCRIPTION
**Pull request by jeswan**
This test is meant as a sanity test for the jiant pipeline and increases code coverage significantly (+23.2%). The test downloads, tokenizes, trains, and evaluates on a dataset. It runs in 31.2 seconds. The test is parameterized so is can potentially be run with different task/model combinations although this would only be possible on the NYU cluster. COPA was chosen as the task for the test since it is small (it can be changed). bert-base-cased was chosen for the model since it is one of the smaller models we support (also can be changed).

The test runs as part of the default test suite with with the `pytest` command.

Note: Includes `nlp` version bump to 0.4.0

### CODE REVIEW

File: tests/proj/simple/test_runscript.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import pytest
+import argparse
+
+from jiant.proj.simple import runscript as run
+import jiant.scripts.download_data.runscript as downloader
+import jiant.utils.zconf as zconf
```
Comment (jeswan) [inline tests/proj/simple/test_runscript.py:6]:
```suggestion
```

File: tests/proj/simple/test_runscript.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import pytest
+import argparse
```
Comment (jeswan) [inline tests/proj/simple/test_runscript.py:2]:
```suggestion
```

File: jiant/proj/simple/runscript.py [hunk 3]
```diff
@@ -34,7 +34,7 @@ class RunConfiguration(zconf.RunConfig):
     # === Misc parameters === #
     train_batch_size = zconf.attr(type=int, default=32)
     max_seq_length = zconf.attr(type=int, default=256)
-    num_train_epochs = zconf.attr(type=int, default=3)
+    num_train_epochs = zconf.attr(type=float, default=3)
```
Comment (jeswan) [inline jiant/proj/simple/runscript.py:37]:
```suggestion
    num_train_epochs = zconf.attr(type=int, default=3)
```

**Comment by zphang** [comment]
Looks fine to me. Also FYI, RunConfiguration objects can be constructed like regular Python objects, so no need to use cl_args.

**Comment by jeswan** [comment]
@zphang Ah, didn't know that. Will make that change!

File: jiant/proj/simple/runscript.py [hunk 4]
```diff
@@ -102,11 +103,18 @@ def run_simple(args: RunConfiguration):
...
         # === Step 1: Write task configs based on templates === #
         full_task_name_list = sorted(list(set(args.train_tasks + args.val_tasks + args.test_tasks)))
-        task_config_path_dict = create_and_write_task_configs(
-            task_name_list=full_task_name_list,
-            data_dir=args.data_dir,
-            task_config_base_path=os.path.join(args.exp_dir, "task_configs"),
-        )
+        task_config_path_dict = {}
+        if args.create_config:
+            task_config_path_dict = create_and_write_task_configs(
```
Comment (zphang) [inline jiant/proj/simple/runscript.py:108]: I think we should deprecate `create_and_write_task_configs` and just use downloader configs from here on.

File: tests/proj/simple/test_runscript.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import os
+import pytest
+
+import jiant.utils.python.io as py_io
+from jiant.proj.simple import runscript as run
+import jiant.scripts.download_data.runscript as downloader
+
+
```
Comment (zphang) [inline tests/proj/simple/test_runscript.py:8]: Mind adding a comment here (or documentation elsewhere) for what command to use to run this?

Comment (jeswan) [reply tests/proj/simple/test_runscript.py:8]: This is run by default with the usual `pytest` command. No special command is necessary. What are your thoughts on what needs to be clarified?

File: jiant/proj/simple/runscript.py (hunk 4 above)
Comment (jeswan) [reply jiant/proj/simple/runscript.py:108]: Out of scope for these changes but I agree. Opened an issue here: https://github.com/jiant-dev/jiant/issues/170

File: tests/proj/simple/test_runscript.py (hunk 5 above)
Comment (zphang) [reply tests/proj/simple/test_runscript.py:8]: I see, I got confused with the slow tests.

**Review by zphang — APPROVED** [review]

**Pull request merged**